app/services/content_renderer.rb

```diff
@@ -6,51 +6,32 @@ class ContentRenderer
   class ContentParsingError < StandardError
   end
 
-  delegate :calculate_reading_time, to: :processed
-  delegate :content, :front_matter, to: :parsed_input
-
   attr_reader :input, :source, :user
+  attr_accessor :reading_time, :front_matter
 
   def initialize(input, source:, user:)
     @input = input || ""
     @source = source
     @user = user
   end
 
-  def processed
-    @processed ||= processor.new(content, source: source, user: user)
-  # TODO: Replicating prior behaviour, but this swallows errors we probably shouldn't
-  rescue StandardError => e
-    raise ContentParsingError, e.message
-  end
-
-  def finalize(link_attributes: {})
+  def process(link_attributes: {}, calculate_reading_time: false)
+    fixed = fixer.call(input)
+    parsed = front_matter_parser.call(fixed)
+    self.front_matter = parsed.front_matter
```

Comment:
I'm not fan of processing and returning the processed result AND modifying state (setting `front_matter` and `reading_time` here), but splitting code into multiple methods makes it harder to read and modify here. We'll see how to change it while working on extracting `DisplayAd` rendering.